roles_n_rules/switch.py

```diff
@@ -33,25 +34,25 @@ def pay():
                     break
                 
                 db_set(user_id,'uses',i)
+                answer.msg(power.power(user_role),db_get(user_id,'channel'))
                 break
 
         # Force Cupid to fall in love
         if user_role == "Cupid" and db_get(user_id,'uses') > 0:
             chosen = False
             attempts = 0
 
-            while not chosen and attempts < 100:
+            while not chosen and attempts < 1000:
```

Comment:
where does attempts get updated. to me it looks like you never increment attempts.. also you might want to use a for loop like this
```py
for _ in range(1000):
    # do stuff
    if chosen:
        break
```

Reply:
We could, but this can get very heavy memory-wise. I figured it would be easier to keep track of once variable that keeps getting added if it fails.

Reply:
what? do you even KNOW python? first: the size, even if it was all in memory at once would be 9112 bytes. i don't about you, but this doesn't seem like too much.. but even if: the size of a range ITERATOR, which this is, is 48 bytes. 48 bytes. seems like a fucken lot of memory...